components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/OpenShiftGatewayImpl.java

```diff
@@ -0,0 +1,258 @@
+                        "InputStream", e, ExceptionCodes.TEMPLATE_LOAD_EXCEPTION);
+
+            }
+        }
+    }
+
+    /**
+     * @see ContainerBasedGatewayGenerator#createContainerBasedDeployment(String deploymentTemplate, String apiId,
+     * String deploymentName, String namespace, String label)
+     */
+    @Override
+    public void createContainerBasedDeployment(String deploymentTemplate, String apiId, String deploymentName,
+                                               String namespace, String label) throws GatewayException {
+
+        if (masterURL != null) {
+
+            System.setProperty("kubernetes.auth.tryKubeConfig", "false");
+            System.setProperty("kubernetes.auth.tryServiceAccount", "true");
+
+            Config config = new ConfigBuilder().withMasterUrl(masterURL)
+                    .withClientCertFile(certFile).withOauthToken(ContainerBasedGatewayUtils
+                            .getServiceAccountAccessToken()).build();
+            KubernetesClient kubernetesClient = new DefaultKubernetesClient(config);
+            OpenShiftClient client = kubernetesClient.adapt(OpenShiftClient.class);
+             List<HasMetadata> resources;
+
+            try (InputStream inputStream = IOUtils.toInputStream(deploymentTemplate)) {
+                resources = client.load(inputStream).get();
+
+                if (resources.get(0) == null) {
+                    client.close();
+                    throw new GatewayException("No resources loaded from definition : ",
+                            ExceptionCodes.NO_RESOURCE_LOADED_FROM_DEFINITION);
+                }
+
+                HasMetadata resource = resources.get(0);
+                if (resource instanceof Deployment) {
+                    Deployment result =
+                            client.extensions().deployments().inNamespace(namespace).load(inputStream).create();
+                    log.info("Created deployment : ", result.getMetadata().getName());
+
+                } else {
+                    throw new GatewayException("Loaded Resource is not a Kubernetes Deployment ! " + resource,
+                            ExceptionCodes.LOADED_RESOURCE_IS_NOT_VALID);
+                }
+
+            } catch (KubernetesClientException | IOException e) {
+                throw new GatewayException("Client cannot load the file for deployment " + deploymentName + "as an " +
+                        "InputStream", e, ExceptionCodes.TEMPLATE_LOAD_EXCEPTION);
+
+            }
+        }
+    }
+
+    /**
+     * @see ContainerBasedGatewayGenerator#removeContainerBasedGateway(String label, String apiId, String namespace)
+     */
+    @Override
+    public void removeContainerBasedGateway(String label, String apiId, String namespace) throws GatewayException {
+
+        if (masterURL != null) {
+            System.setProperty("kubernetes.auth.tryKubeConfig", "false");
```

Comment:
cant we move client initlaization to single place? I see lots of places which do the same